projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+
+using RabbitMQ.Client.Events;
+using RabbitMQ.Client.Framing.Impl;
+
+namespace RabbitMQ.Client
+{
+    internal class RabbitMQActivitySource
+    {
+        internal static ActivitySource source = new ActivitySource("RabbitMQ.Client", typeof(RabbitMQActivitySource).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion);
+        private static readonly TextMapPropagator Propagator = new CompositeTextMapPropagator(new TextMapPropagator[] { new TraceContextPropagator(), new BaggagePropagator() });
+
+        static RabbitMQActivitySource()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+        }
+
+        internal static Activity Send(BasicPublish basicPublish, int bodySize)
+        {
+            if (source.HasListeners())
+            {
+                Activity activity = StartRabbitMQActivity($"{basicPublish._routingKey} send", ActivityKind.Producer, Activity.Current?.Context ?? default);
+                if (activity != null && activity.IsAllDataRequested)
+                {
+                    PopulateMessagingTags("send", basicPublish._routingKey, basicPublish._exchange, 0, null, bodySize, activity);
```

Comment:
The "send" here goes into `messaging.operation`
>  If the operation is “send”, this attribute MUST NOT be set, since the operation can be inferred from the span kind in that case.

You can send `null` here. The OTel API will remove the tag if its value is `null`

Reply:
This seems to only apply to message consumers. This should be set for producers/clients (this is for producers): https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#operation-names